### PR TITLE
Enable Locale Selection in Dynamo Preferences Panel

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

This PR enables Locale selection in Dynamo Preferences Panel. As a result, Dynamo will be localized in the new locale after a restart. Notice that the selection does not affect current Dynamo session.

User experience: (notice the new language selections resources are in DynamoCore not localized yet)
New info icon added
image

Toast notification added
image

Gif:
ChangeLocale

Serialization changes:
image

TODO:
- [x] Try toast notification and add information icon

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Enable Locale Selection in Dynamo Preferences Panel

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/DynamoCoreTests/Configuration/PreferenceSettingsTests.cs [hunk 1]
```diff
@@ -398,7 +401,7 @@ public void TestImportCopySettings()
             string failMessage = $"The file {newSettingslFilePath} exist: {newSettingsExist.ToString()} | DiffProps: {diffProps.ToString()} | TotProps: {totProps.ToString()} | Default Sett NumberFormat: {defSettNumberFormat} | New Sett NumberFormat: {newSettNumberFormat} | First Property with the same value {firstPropertyWithSameValue}";
 
             // checking if the new Setting are completely different from the Default
-            Assert.IsTrue(checkDifference.DifferentPropertyValues.Count == checkDifference.Properties.Count, failMessage);
+            Assert.IsTrue(checkDifference.DifferentPropertyValues.Count == checkDifference.DifferentPropertyValues.Count, failMessage);
```
[the remark this answers is not in the document]
Comment (QilongTang) [reply test/DynamoCoreTests/Configuration/PreferenceSettingsTests.cs:404]: Thanks, I forgot to revert this change before merging. Let me do that

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs [hunk 2]
```diff
@@ -161,24 +161,16 @@ public string SelectedLanguage
...
             {
-                selectedLanguage = value;
-                RaisePropertyChanged(nameof(SelectedLanguage));
-            }
-        }
-
-        /// <summary>
-        /// Controls the Selected option in Node Font Size ComboBox
-        /// </summary>
-        public string SelectedFontSize
```
Comment (reddyashish) [inline src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:172]: So there is no font-size setting in the preferences panel?

**Review by reddyashish — APPROVED** [review]
LGTM

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:172]: Hi @reddyashish We removed such a setting. There was a font size setting when we started Preferences Panel project and recently we confirmed with customers that such a setting would come with no major value. Customers would rely on workspace zoom level after all the improvements we did. So I had a PR to remove the UI, now in this PR I am removing the underneath binding code as well..

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 3]
```diff
@@ -757,6 +762,7 @@ public PreferenceSettings()
             // Default Settings
             IsFirstRun = true;
             IsAnalyticsReportingApproved = true;
+            Locale = Configurations.SupportedLocaleDic.FirstOrDefault().Value;
```
Comment (RobertGlobant20) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:765]: Configurations.SupportedLocaleDic can be empty or null?

**Review by RobertGlobant20 — APPROVED** [review]
LGTM with one comment

File: src/DynamoCore/Configuration/PreferenceSettings.cs (hunk 3 above)
Comment (QilongTang) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:765]: Nope, it is initialized as constants

File: src/DynamoCore/Configuration/IPreferences.cs [hunk 4]
```diff
@@ -146,6 +146,11 @@ public interface IPreferences
         /// <returns>The active state</returns>
         bool GetIsBackgroundPreviewActive(string name);
 
+        /// <summary>
+        /// Return the locale of Dynamo UI
+        /// </summary>
+        string Locale { get; set; }
```
Comment (mjkkirschner) [inline src/DynamoCore/Configuration/IPreferences.cs:152]: this is an API break - don't modify existing interfaces!

**Pull request merged**